From b59ab93465b2aa029bfa7e7ece1391ae5d763eb8 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:08 -0400
Subject: [block] remove dead code in scsi_ioctl:blk_verify_command

Message-id: <1402670238-13916-21-git-send-email-snitzer@redhat.com>
Patchwork-id: 83772
O-Subject: [RHEL7.1 PATCH 020/150] block: remove dead code in scsi_ioctl:blk_verify_command
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit da52f22fa924b4a21d8e11fbfd3eeebd7a90a366
Author: Dave Jones <davej@redhat.com>
Date:   Thu May 29 15:11:30 2014 -0400

    block: remove dead code in scsi_ioctl:blk_verify_command

    filter gets assigned the address of blk_default_cmd_filter on
    entry to this function, so the !filter condition can never be true.

    Signed-off-by: Dave Jones <davej@redhat.com>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/scsi_ioctl.c b/block/scsi_ioctl.c
index b379ac3..10173c0 100644
--- a/block/scsi_ioctl.c
+++ b/block/scsi_ioctl.c
@@ -208,10 +208,6 @@ int blk_verify_command(struct request_queue *q,
  if (capable(CAP_SYS_RAWIO) || blk_queue_unpriv_sgio(q))
   return 0;
 
- /* if there's no filter set, assume we're filtering everything out */
- if (!filter)
-  return -EPERM;
-
  /* Anybody who can open the device can do a read-safe command */
  if (test_bit(cmd[0], filter->read_ok))
   return 0;
-- 
1.7.1